client/packages/system/src/Stock/Components/StockLineEditModal.tsx

```diff
@@ -72,6 +83,14 @@ export const StockLineEditModal: FC<StockLineEditModalProps> = ({
     },
   ];
 
+  const onChange = () => setHasChanged(true);
+
+  useEffect(() => {
+    addEventListener('onChange', 'StockEditForm', onChange);
```

Comment:
adding this so that the validation is updated when the plugin changes it's own data

Reply:
the `onCancel` is vaguely what I had in mind for that, it's slightly different than unchanging tho. I couldn't see a need to 'unChange' yet, and I've only implemented the minimum of what I think is useful